### PR TITLE
[DataFrame] Implement ray.DataFrame.drop w/ tests

### PR DESCRIPTION
**Pull request by SaladRaider**
## What do these changes do?

Implements the ray.Dataframe.drop method. Checks if values of labels, index or columns exist in the pandas DataFrame partitions, if the errors flag is set in order to raise the same errors that pandas does. @devin-petersohn

### CODE REVIEW

**Comment by robertnishihara** [comment]
ok to test

File: python/ray/dataframe/dataframe.py [hunk 1]
```diff
@@ -463,7 +476,47 @@ def dot(self, other):
 
     def drop(self, labels=None, axis=0, index=None, columns=None, level=None,
              inplace=False, errors='raise'):
-        raise NotImplementedError("Not Yet implemented.")
+        if errors == 'raise':
```
Comment (devin-petersohn) [inline python/ray/dataframe/dataframe.py]: We will wait on index and the changes should be easier.

File: python/ray/dataframe/dataframe.py [hunk 2]
```diff
@@ -134,6 +134,19 @@ def _map_partitions(self, func, *args):
 
         return DataFrame(new_df, self.columns)
 
+    def _map_partitions_to_values(self, func, *args):
```
Comment (devin-petersohn) [inline python/ray/dataframe/dataframe.py]: I prefer to just use `_map_partitions(...)._df` for now.

**Comment by devin-petersohn** [comment]
Closing in favor of #1544. @SaladRaider feel free to reopen if you think it shouldn't be closed.

**Pull request closed without merging**